kafka-rest/src/main/java/io/confluent/kafkarest/v2/KafkaConsumerManager.java

```diff
@@ -503,6 +600,74 @@ private synchronized void updateExpiration(KafkaConsumerState state) {
     Consumer createConsumer(Properties props);
   }
 
+  private class ReadTaskState {
+    final KafkaConsumerReadTask task;
+    final KafkaConsumerState consumerState;
+    final ReadCallback callback;
+
+    public ReadTaskState(KafkaConsumerReadTask task,
+                         KafkaConsumerState state,
+                         ReadCallback callback) {
+
+      this.task = task;
+      this.consumerState = state;
+      this.callback = callback;
+    }
+  }
+
+  private class ReadTaskSchedulerThread extends Thread {
+    AtomicBoolean isRunning = new AtomicBoolean(true);
+    CountDownLatch shutdownLatch = new CountDownLatch(1);
+
+    ReadTaskSchedulerThread() {
+      super("Read Task Scheduler Thread");
+      setDaemon(true);
+    }
+
+    @Override
+    public void run() {
+      try {
+        long timeoutMs = 25;
+        while (isRunning.get()) {
+          RunnableReadTask readTask = delayedReadTasks.poll(timeoutMs, TimeUnit.MILLISECONDS);
+          if (readTask != null) {
+            delayedTasks.put(readTask, false);
+            executor.submit(readTask);
+            continue;
+          }
+
+          readTask = delayedReadTasks.peek();
```

Comment:
I'm not sure I understand why we need this logic. The call to `DelayedQueue.poll(timeout)` will return as soon as the next delay has expired. I think it would be reasonable to poll with a fixed timeout like 500ms just so we're fairly responsive to shutdown. We could make the timeout even longer if we have a nice way to wakeup the delay queue (such as the poison pill I mentioned offline)

Reply:
derp. right. forgot to remove this. Makes total sense to just have a larger timeout in the poll